### PR TITLE
Expose connection pooling and load balancing for grpc clients

### PR DESCRIPTION
**Pull request by teodor-pripoae**

### CODE REVIEW

**Comment by adleong** [comment]
Hi @teodor-pripoae, can you give a little more background on what you'd like to accomplish in this PR.  This change adds the `withSessionPool` method to Client but doesn't expose any way to invoke it.

**Comment by teodor-pripoae** [comment]
Hi,

I'm trying to use the grpc client outside linkerd, from scala code. I've asked on the slack channel about configuring grpc clients with load balancing and session pooling and @olix0r told me that currently it can be configured using `configure` method but will accept a PR which exposes this functionality further.

Currently I've been able to set it like this after exposing this 2 traits:

```scala
 val h2Settings = H2.client
      .withLabel(name)
      .withRequestTimeout(timeout)
      .withRetryBackoff(retryBackoff)
      .withRetryBudget(retryBudget)
      .withLoadBalancer(buildLoadBalancer(config))
      .withSessionPool
      .minSize(poolMin)
      .withSessionPool
      .maxSize(poolMax)
      .withSessionPool
      .maxWaiters(maxWaitersPerHost)
val client = h2Settings.newClient(dest)
```

File: finagle/h2/src/main/scala/com/twitter/finagle/buoyant/H2.scala [hunk 1]
```diff
@@ -11,6 +11,8 @@ import com.twitter.finagle.stack.nilStack
 import com.twitter.finagle.transport.Transport
 import com.twitter.util.{Closable, Future}
 import java.net.SocketAddress
+
+import com.twitter.finagle.param.{WithConcurrentLoadBalancer, WithSessionPool}
 import io.netty.handler.codec.http2.Http2Frame
```
Comment (olix0r) [inline finagle/h2/src/main/scala/com/twitter/finagle/buoyant/H2.scala]: imports should be sorted alphabetically with no spaces

**Comment by teodor-pripoae** [comment]
@olix0r I've updated the PR.

**Comment by adleong** [comment]
@olix0r could you take another look at this?

**Comment by olix0r** [comment]
Whoops! sorry for dropping the ball on this.  If you resolve the merge conflicts we should be able to pull this.

**Comment by adleong** [comment]
Closing this in favor of https://github.com/linkerd/linkerd/pull/1551

**Pull request closed without merging**